kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ByPassUserTaskServiceIntegrationTest.java

```diff
@@ -94,4 +94,42 @@ public void testProcessWithUserTasks() {
         }
     }
 
+    @Test
+    public void testByPassAuthUserTasks() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK2);
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId > 0);
+        try {
+            List<TaskSummary> taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertTaskList(taskList, 1);
+
+            // Get tasks as potential owner by using the authenticated user
+            taskList = taskClient.findTasksAssignedAsPotentialOwner("", 0, 10);
+            assertTaskList(taskList, 1);
+
+            changeUser(USER_JOHN);
+            taskList = taskClient.findTasksAssignedAsPotentialOwner("", 0, 10);
+            assertTaskList(taskList, 0);
+
+            taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_MARY, 0, 10);
+            assertTaskList(taskList, 0);
+
+            taskList = taskClient.findTasksAssignedAsPotentialOwner("other", 0, 10);
+            assertTaskList(taskList, 1);
+
+        } finally {
+            changeUser(USER_YODA);
+            processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
+        }
+    }
+
+    private void assertTaskList(List<TaskSummary> taskList, int size) {
+        assertNotNull(taskList);
+        assertEquals(size, taskList.size());
+        if (taskList.size() > 0) {
+            TaskSummary taskSummary = taskList.get(0);
+            assertEquals("First task", taskSummary.getName());
```

Comment:
You could pass the name of the task as parameter and refactor other similar asserts in the other test method `testProcessWithUserTasks`, converting task names into constants, wdyt?

Reply:
yes! thanks for the hint!